docs/usage.rst

```diff
@@ -125,3 +125,21 @@ Pre-0.7.0, this is how it worked:
 
 * Cloned cookiecutters were not saved locally.
 
+Update a cookiecutter generated project
+---------------------------------------
+
+If you want to update a project generated by a cookiecutter template, give its PATH to
+the -u option of cookiecutter. This project should have a `.cookiecutter.json` file to
+interpret the rendering done before.
+
+For now this option only works with git project.
+
+A branch named `cookiecutter` should be present and has its HEAD on the last cookiecutter
+update, or the commit that introduced cookiecutter changes.
+It is recommended to be have this branch as close as possible from the cookiecutter template
+rendered, to avoid any conflicts at this point.
+
+The update option will bring the new changes on this branch, then try to bring these changes
```

Comment:
Possible wording changes: "The update option will apply the new changes on this branch, then try to merge these changes to the `master` branch."

Also, our team uses feature branches, so we rarely if ever make changes directly to `master`. Is there a way to make the changes to another, user-specified branch instead?